content/en/service_management/app_builder/queries.md

```diff
@@ -0,0 +1,149 @@
+
+{{< code-block lang="js" >}}${select0.value && select0.value.length > 0}{{< /code-block >}}
+
+## Post-query transformation
+
+Perform a post-query transformation to simplify or transform the output of a query. Add a post-query transformation in the **Advanced** section of a query.
+
+For example, the Slack _List Channels_ action returns an array of dictionaries containing the ID and name for each channel. To discard the IDs and return only an array of names, add the following query transformation:
+
+{{< code-block lang="js" collapsible="false" >}}
+// Use `outputs` to reference the query's unformatted output.
+// TODO: Apply transformations to the raw query output
+arr = []
+object = outputs.channels
+for (var item in object) {
+    arr.push(object[item].name);
+}
+
+return arr
+{{< /code-block >}}
+
+## Post-query hooks
+
+Similar to UI component events, you can configure a reaction to trigger after a query executes. A **post-query hook** can set a UI component state, open or close a modal, trigger another query, or even run custom JavaScript. For example, the [ECS Task Manager][7] blueprint's `scaleService` query uses a post-query hook to rerun the `describeService` query after it executes.
+
+## Error notifications
+
+To display a toast (a brief notification message) to the user when the system returns an error, toggle **Show Toast on Errors** in the **Advanced** section of a query.
+
+## Confirmation prompts
+
+To prompt a user for confirmation before the query runs, toggle the **Requires Confirmation** option in the **Advanced** section of a query.
+
+## Example app
+
+### Return workflow results to an app
+App Builder queries can trigger Workflow Automation workflows. Apps can then use the results of those workflows.
+
+This app triggers a workflow, which sends a poll to a Slack channel asking the user to pick from one of two options. Based on the option the user chooses, the workflow issues one of two different HTTP GET requests, which then returns data that is displayed in the app.
+
+{{< img src="service_management/app_builder/workflow-trigger-from-app.mp4" alt="Clicking Trigger Workflow polls Slack then returns a random cat or dog fact" video="true" width="486px">}}
+
+{{% collapse-content title="Build the app" level="h4" %}}
+
+##### Create workflow
+
+1. In a new workflow canvas, under **Datadog Triggers**, click **App**. 
+1. Under the **App** trigger step, click the plus (**+**) icon then search for "Make a decision" and select the **Make a decision** Slack action.
+1. Select your workspace and choose a channel to poll.
+1. Fill in the prompt text "Cat fact or dog fact?" and change the button choices to "Cat fact" and "Dog fact".
```

Comment:
```suggestion
1. Fill in the prompt text with "Cat fact or dog fact?" and change the button choices to "Cat fact" and "Dog fact".
```

Reply:
I actually think the original is more correct.

Reply:
I don't feel strongly about my edit and can understand your original, too. 😄